src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -417,6 +417,10 @@ private BoundForEachStatement BindForEachPartsWorker(DiagnosticBag diagnostics,
 
             var foreachKeyword = _syntax.ForEachKeyword;
             ReportDiagnosticsIfObsolete(diagnostics, getEnumeratorMethod, foreachKeyword, hasBaseReceiver: false);
+            ReportDiagnosticsIfUnmanagedCallersOnly(diagnostics, getEnumeratorMethod, foreachKeyword.GetLocation());
```

Comment:
Just to clarify, only the `GetEnumerator` could be an extension method?

Reply:
That's correct.

---
In reply to: [481412634](https://github.com/dotnet/roslyn/pull/47311#discussion_r481412634) [](ancestors = 481412634)